packages/core/test/utils.spec.ts

```diff
@@ -21,6 +21,11 @@ describe('utils', () => {
       expect(deepMerge({ a: 1 }, { a: undefined })).toEqual({ a: 1 })
     })
 
+    it('should support null values', () => {
+      // tslint:disable-next-line: no-null-keyword
+      expect(deepMerge({ a: {} }, { a: null })).toEqual({ a: null })
```

Comment:
There is a strange behavior:
```ts
expect(deepMerge({ a: 1 }, undefined)).toEqual({ a: 1 })
expect(deepMerge({ a: 1 }, null)).toEqual(null)
```
could it be better to have:
```ts
expect(deepMerge({ a: 1 }, null)).toEqual({ a: 1 })
```

Reply:
Mmh, if we ignore `null` values, that would make things like:

```ts
DD_LOGS.setLogsGlobalContext({ page: 'toto' })
DD_LOGS.logger.log('message', { page: null })
// context is { page: 'toto' }
```
I find this a bit confusing. I can understand that `undefined` gets ignored, as it's kind of idiomatic in JS:

```ts
function log(page = 'toto') {
   // [...]
}
log(undefined) // page is 'toto'
```
But it's not the case for null values

```ts
log(null) // page is null
```

But we can also consider the `??` operator which handle `null` and `undefined` values the same:

```js
null ?? 1 // 1
undefined ?? 1 // 1
```

I have mixed feelings about this, but I think it's fine if we handle `null` and `undefined` values differently and don't ignore `null` values, so the user can override its context with `null`. WDYT?

Reply:
I would be in favor of having the same behavior as [lodash](https://lodash.com/docs/4.17.15#merge):
<img width="228" alt="Capture d’écran 2020-09-25 à 11 53 36" src="https://user-images.githubusercontent.com/1331991/94253604-d58ecd80-ff25-11ea-927b-498198d29b2c.png">

Reply:
I see. I'll give it a shot